### PR TITLE
build: upgrade etcd-client to 0.12.2

### PR DESCRIPTION
**Pull request by tisonkun**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

* https://github.com/etcdv3/etcd-client/pull/68
* https://github.com/GreptimeTeam/greptimedb/pull/2734#issuecomment-1813785955

Following I'll create a few follow-ups to improve the code and prepare to add chroot for etcd connection.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -82,7 +82,7 @@ datafusion-physical-expr = { git = "https://github.com/apache/arrow-datafusion.g
 datafusion-sql = { git = "https://github.com/apache/arrow-datafusion.git", rev = "26e43acac3a96cec8dd4c8365f22dfb1a84306e9" }
 datafusion-substrait = { git = "https://github.com/apache/arrow-datafusion.git", rev = "26e43acac3a96cec8dd4c8365f22dfb1a84306e9" }
 derive_builder = "0.12"
-etcd-client = "0.12"
+etcd-client = "0.12.2"
```
Comment (tisonkun) [inline Cargo.toml:85]: The upstream bumps patch version only for new features. We'd better specify the patch version to avoid unexpected upragde.

**Comment by tisonkun** [comment]
cc @MichaelScofield

File: rust-toolchain.toml [hunk 2]
```diff
@@ -1,2 +1,2 @@
 [toolchain]
-channel = "nightly-2023-08-07"
+channel = "nightly-2023-10-21"
```
Comment (tisonkun) [inline rust-toolchain.toml:2]: See -

* https://github.com/sgodwincs/dlv-list-rs/pull/19
* https://github.com/risingwavelabs/risingwave/pull/12992

File: Cargo.toml (hunk 1 above)
Comment (sunng87) [reply Cargo.toml:85]: Actually in cargo `0.12.2` means any version `<0.13, >= 0.12.2`. If we really want fixed version of etcd-client, we will need to pin `=0.12.2` but that's not best practice.

Comment (tisonkun) [reply Cargo.toml:85]: OK. So let's revert to `0.12`?

Comment (tisonkun) [reply Cargo.toml:85]:
```suggestion
etcd-client = "0.12"
```

If so, please help apply this suggestion - I'll be back tomorrow morning :D

**Review by MichaelScofield — APPROVED** [review]

**Comment by tisonkun** [comment]
cc @fengjiachun

File: rust-toolchain.toml (hunk 2 above)
Comment (shuiyisong) [reply rust-toolchain.toml:2]: How about upgrade to the latest nightly :D

Comment (tisonkun) [reply rust-toolchain.toml:2]: I'm trying to minimize the change. We can try to upgrade it in another PR but this one only include the most necessary ones.

**Review by fengjiachun — APPROVED** [review]
LGTM

**Pull request merged**